core/src/main/scalajvm/sttp/client4/httpclient/HttpClientSyncBackend.scala

```diff
@@ -1,48 +1,114 @@
     customEncodingHandler: SyncEncodingHandler
-) extends HttpClientBackend[Identity, Nothing, Any, InputStream](
+) extends HttpClientBackend[Identity, Nothing, WebSockets, InputStream](
       client,
       closeClient,
       customEncodingHandler
     )
-    with SyncBackend {
+    with WebSocketSyncBackend {
 
+  private implicit val ec: ExecutionContext = ExecutionContext.global
   override val streams: NoStreams = NoStreams
 
-  override def send[T](request: GenericRequest[T, R]): Response[T] =
-    adjustExceptions(request) {
-      val jRequest = customizeRequest(convertRequest(request))
-      val response = client.send(jRequest, BodyHandlers.ofInputStream())
-      readResponse(response, Left(response.body()), request)
-    }
+  override protected def sendRegular[T](request: GenericRequest[T, R]): Identity[Response[T]] = {
+    val jRequest = customizeRequest(convertRequest(request))
+    val response = client.send(jRequest, BodyHandlers.ofInputStream())
+    readResponse(response, Left(response.body()), request)
+  }
+
+  override protected def sendWebSocket[T](request: GenericRequest[T, R]): Identity[Response[T]] = {
+    val queue = createSimpleQueue[WebSocketEvent]
+    val sequencer = createSequencer
+    sendWebSocket(request, queue, sequencer).handleError {
+      case e: CompletionException if e.getCause.isInstanceOf[WebSocketHandshakeException] =>
+        readResponse(
+          e.getCause.asInstanceOf[WebSocketHandshakeException].getResponse,
+          Left(emptyInputStream()),
+          request
+        )
+    }(monad)
+  }
 
-  override def monad: MonadError[Identity] = IdMonad
+  private def sendWebSocket[T](
+      request: GenericRequest[T, R],
+      queue: SimpleQueue[Identity, WebSocketEvent],
+      sequencer: Sequencer[Identity]
+  ): Identity[Response[T]] = {
+    val isOpen: AtomicBoolean = new AtomicBoolean(false)
+    val responseCell = new ArrayBlockingQueue[Either[Throwable, Future[Response[T]]]](1)
 
-  private def adjustExceptions[T](request: GenericRequest[_, R])(t: => T): T =
-    SttpClientException.adjustExceptions(monad)(t)(
-      SttpClientException.defaultExceptionToSttpClientException(request, _)
+    def fillCellError(t: Throwable): Unit = responseCell.add(Left(t))
+
+    def fillCell(wr: Future[Response[T]]): Unit = responseCell.add(Right(wr))
+
+    val listener = new DelegatingWebSocketListener(
+      new AddToQueueListener(queue, isOpen),
+      ws => {
+        val webSocket = new WebSocketImpl[Identity](
+          ws,
+          queue,
+          isOpen,
+          sequencer,
+          monad,
+          cf =>
+            monad.suspend {
+              Try(cf.get()).fold(monad.error, _ => monad.unit(()))
+            }
+        )
+        val baseResponse = Response((), StatusCode.SwitchingProtocols, "", Nil, Nil, request.onlyMetadata)
+        val body = Future(blocking(bodyFromHttpClient(Right(webSocket), request.response, baseResponse)))
```

Comment:
since we're in a blocking world I think we can simply do `Future.successfull(bodyFrom...)`